### ISSUE TITLE
Exception in grid CellSizeAndPositionManager

### ISSUE THREAD

**Issue by prog8** [bvaughn/react-virtualized#721]
Hi. I'm fighting with the exception which is thrown from time to time in Grid.

```javascript
CellSizeAndPositionManager.js:184 Uncaught TypeError: Cannot read property 'offset' of undefined
    at CellSizeAndPositionManager.getVisibleCellRange (CellSizeAndPositionManager.js:184)
    at ScalingCellSizeAndPositionManager.getVisibleCellRange (ScalingCellSizeAndPositionManager.js:150)
    at Grid._calculateChildrenToRender (Grid.js:523)
    at Grid.componentWillUpdate (Grid.js:403)
    at ReactCompositeComponentWrapper._performComponentUpdate (ReactCompositeComponent.js:701)
    at ReactCompositeComponentWrapper.updateComponent (ReactCompositeComponent.js:634)
    at ReactCompositeComponentWrapper.receiveComponent (ReactCompositeComponent.js:534)
    at Object.receiveComponent (ReactReconciler.js:131)
    at ReactCompositeComponentWrapper._updateRenderedComponent (ReactCompositeComponent.js:737)
    at ReactCompositeComponentWrapper._performComponentUpdate (ReactCompositeComponent.js:715)
```

I looked into details (though I don't know whole logic) and I realized that in the [line](https://github.com/bvaughn/react-virtualized/blob/master/source/Grid/utils/CellSizeAndPositionManager.js#L244) there is check if low > 0 and only return low-1 in such a case. What if low is 0 which seems to be a [common case](https://github.com/bvaughn/react-virtualized/blob/master/source/Grid/utils/CellSizeAndPositionManager.js#L290)? It seems for such scenarios we have nice `undefined` returned from binary search.

**Comment by bvaughn** [comment]
Don't believe I've ever seen the error you're referring to. Can you provide a repro?

**Comment by prog8** [comment]
I don't exactly know in what circumstances this happens. Something when I change data in Grid from many rows to 1 row (can't guarantee that in meantime there are no 0 rows). I only know that when I return 0 instead of undefined (https://github.com/bvaughn/react-virtualized/blob/master/source/Grid/utils/CellSizeAndPositionManager.js#L244) the error doesn't happen

**Comment by prog8** [comment]
I have one more important information. Maybe this is the problem actually. I realized that I set rowCount to 1 and provide Grid with 50 actual rows (see screenshot below to see what values CellSizeAndPositionManager sees).
After I fixed this (give grid correct number of total rows) I cannot reproduce the problem. I am not sure if this case should be checked inside Grid or not (user should make sure he doesn't provide Grid with misleading values). Maybe this should be on the user side.

![selection_148](https://user-images.githubusercontent.com/6692291/27902776-34cb9aba-6237-11e7-9e66-d3586026da41.png)
